src/main/java/com/powsybl/openloadflow/sensi/DcSensitivityAnalysis.java

```diff
@@ -667,6 +667,24 @@ public void calculateContingencySensitivityValues(PropagatedContingency continge
             DenseMatrix newFactorStates = factorStates;
             boolean participatingElementsChanged = (isDistributedSlackOnGenerators(lfParameters) && !contingency.getGeneratorIdsToLose().isEmpty())
                     || (isDistributedSlackOnLoads(lfParameters) && !contingency.getLoadIdsToShift().isEmpty());
+            boolean rhsChanged = false;
+            if (hasMultiVariables) {
+                Set<LfBus> affectedBuses = new HashSet<>();
+                for (var e : contingency.getLoadIdsToShift().entrySet()) {
+                    String busId = e.getKey();
+                    LfBus bus = lfNetwork.getBusById(busId);
+                    if (bus != null) {
+                        affectedBuses.add(bus);
+                    }
+                }
+                for (String id : contingency.getGeneratorIdsToLose()) {
+                    LfBus bus = lfNetwork.getGeneratorById(id).getBus();
+                    if (bus != null) {
+                        affectedBuses.add(bus);
+                    }
+                }
+                rhsChanged = rescaleGlsk(factorGroups, affectedBuses);
```

Comment:
Are you sure, in case of multiple contingencies that we don't need to reset the weights as for a particular contingency part of the GLSK we update it but for next contingency we should restart from initial weights?

Reply:
Yes, but it is what is done, no? In the factor, we keep them.